From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jolla.com>
Date: Tue, 7 Jan 2025 13:40:35 +0200
Subject: [PATCH] Disable gtk-doc

---
 autogen.sh   | 20 ++++++++++++++++++++
 configure.ac |  6 ++++++
 2 files changed, 26 insertions(+)

diff --git a/autogen.sh b/autogen.sh
index a47248cc4..7d67f8e33 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -17,9 +17,29 @@ elif automake-1.11 --version >/dev/null 2>/dev/null && \
     export ACLOCAL
 fi
 
+# check if gtk-doc is explicitly disabled
+for ag_option in $@
+do
+  case $ag_option in
+    -disable-gtk-doc | --disable-gtk-doc)
+    enable_gtk_doc=no
+  ;;
+  esac
+done
+
 (
     cd "$srcdir"
+if test x$enable_gtk_doc = xno; then
+    if test -f gtk-doc.make; then :; else
+       echo "EXTRA_DIST = missing-gtk-doc" > gtk-doc.make
+    fi
+    echo "WARNING: You have disabled gtk-doc."
+    echo "         As a result, you will not be able to generate the API"
+    echo "         documentation and 'make dist' will not work."
+    echo
+else
     gtkdocize
+fi
     autoreconf -i -f
 )
 
diff --git a/configure.ac b/configure.ac
index ea5bbce60..70aec98a6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -58,7 +58,13 @@ AC_PROG_MKDIR_P
 # pkg-config 0.21 adds CFLAGS from Requires.private, which we need
 PKG_PROG_PKG_CONFIG([0.21])
 
+AM_CONDITIONAL(ENABLE_GTK_DOC, test "x$enable_gtk_doc" = xyes)
+AM_CONDITIONAL(DISABLE_GTK_DOC, test "x$disable_gtk_doc" = xno)
+# Check for GTK_DOC_CHECK availability. The GTK_DOC_CHECK invocation
+# must be on its own line, gtkdocize relies on it
+m4_ifdef([GTK_DOC_CHECK], [
 GTK_DOC_CHECK([1.17],[--flavour no-tmpl])
+])
 
 GOBJECT_INTROSPECTION_CHECK([1.30])
 
