diff --git a/configure.ac b/configure.ac
index a63a0afe9..ce2bb4b9f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -12,7 +12,7 @@ AC_PREREQ([2.60])
 m4_define([tp_glib_major_version], [0])
 m4_define([tp_glib_minor_version], [24])
 m4_define([tp_glib_micro_version], [1])
-m4_define([tp_glib_nano_version], [1])
+m4_define([tp_glib_nano_version], [0])
 
 # If library source has changed since last release, increment revision
 # If interfaces have been added, removed or changed since last release,
