diff --git a/telepathy-glib/channel-contacts.c b/telepathy-glib/channel-contacts.c
index 687e6c5..1316a41 100644
--- a/telepathy-glib/channel-contacts.c
+++ b/telepathy-glib/channel-contacts.c
@@ -733,6 +733,9 @@ _tp_channel_contacts_handle_owners_changed (TpChannel *self,
   if (self->priv->cm_too_old_for_contacts)
       return;
 
+  if (!self->priv->group_contact_owners)
+      return;
+
   g_assert (self->priv->group_contact_owners != NULL);
 
   data = g_slice_new (HandleOwnersChangedData);
diff --git a/telepathy-glib/channel-group.c b/telepathy-glib/channel-group.c
index 204e398..8dff756 100644
--- a/telepathy-glib/channel-group.c
+++ b/telepathy-glib/channel-group.c
@@ -1385,6 +1385,11 @@ _tp_channel_get_group_properties (TpChannel *self)
   if (priv->handle_owners_changed_sig == NULL)
     DIE ("HandleOwnersChanged");
 
+  /* First try the 0.17 API (properties). If this fails we'll fall back */
+  tp_cli_dbus_properties_call_get_all (self, -1,
+      TP_IFACE_CHANNEL_INTERFACE_GROUP, tp_channel_got_group_properties_cb,
+      NULL, NULL, NULL);
+
   priv->handle_owners_changed_detailed_sig =
       tp_cli_channel_interface_group_connect_to_handle_owners_changed_detailed (
           self, tp_channel_handle_owners_changed_detailed_cb, NULL, NULL, NULL,
@@ -1392,11 +1397,6 @@ _tp_channel_get_group_properties (TpChannel *self)
 
   if (priv->handle_owners_changed_detailed_sig == NULL)
     DIE ("HandleOwnersChangedDetailed");
-
-  /* First try the 0.17 API (properties). If this fails we'll fall back */
-  tp_cli_dbus_properties_call_get_all (self, -1,
-      TP_IFACE_CHANNEL_INTERFACE_GROUP, tp_channel_got_group_properties_cb,
-      NULL, NULL, NULL);
 }
 
 G_GNUC_END_IGNORE_DEPRECATIONS
