diff --git a/configure.ac b/configure.ac
index 9ffa365..b589ecd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -194,7 +194,7 @@ dnl docs/Makefile.am needs to know whether it's an out of tree build
 dnl (srcdir != builddir)
 AM_CONDITIONAL([OUT_OF_TREE_BUILD], [test "z$ac_srcdir" != z.])
 
-tpglibtestsdir=${libdir}/telepathy-glib-tests
+tpglibtestsdir=/opt/tests/telepathy-glib
 AC_SUBST(tpglibtestsdir)
 
 dnl Check for Glib
